src/main/scala/dibl/Pattern.scala

```diff
@@ -15,70 +15,74 @@
-    val tag = s"<text style='font-family:Arial;font-size:11pt'>\n" +
-      s"\t <tspan x='${offsetX - 50}' y='${offsetY - 80}'>$tileType, $hXw, $m</tspan>\n" +
-      s"\t <tspan x='${offsetX - 50}' y='${offsetY - 60}' style='fill:#008;'>\n" +
-      s"\t  <a xlink:href='$link'>pair/thread diagrams</a>\n" +
-      s"\t </tspan>\n" +
-      s"\t</text>\n"
-    s"\n<g>\n$clones\n\t$tag\n\t<g id='$groupId'>\n${original(relative)}\t</g>\n</g>\n"
+    s"""
+      |<text style='font-family:Arial;font-size:11pt'>
+      | <tspan x='${offsetX - 50}' y='${offsetY - 80}'>$tileType, $hXw, $m</tspan>
+      | <tspan x='${offsetX - 50}' y='${offsetY - 60}' style='fill:#008;'>
+      |  <a xlink:href='$link'>pair/thread diagrams</a>
+      | </tspan>
+      |</text>"""
   }
 
-  private def clones: String = {
-    val brickOffset = if (tileType == "bricks") cols * 5 else 0
-    def cloneRows(row1: Int): String = {
-      val row2 = row1 + rows * 10 // TODO refactor into TileType
-      List.range(start = -(if (tileType == "bricks")brickOffset else 10*cols), end = 250, step = cols * 10).
-        map(w => {
-          clone(w, row1) + clone(w - brickOffset, row2)
-        }).mkString("")
-    }
-    val cloneAtOriginal = clone(0, 0)
-    List.range(start = -rows * 10, end = 200, step = rows * 20)
-      .map(h => cloneRows(h)).mkString("")
-      .replace(cloneAtOriginal, cloneAtOriginal.replace("#000", "#008"))
-  }
+  def createConnectors(m: M): String = {
+    def createTwoIn(row: Int, col: Int, cell: SrcNodes): String = if (cell.isEmpty) "" else {
 
-  private def clone(i: Int, j: Int): String =
-    s"\t<use transform='translate($i,$j)' xlink:href='#$groupId' ${"style='stroke:#000;fill:none'"}/>\n"
+      val nodeX = col * 10 + offsetX
+      val nodeY = row * 10 + offsetY
+      val (leftRow, leftCol)= cell(0)
+      val (rightRow, rightCol) = cell(1)
 
-  private def original(m: M): String =
-    m.indices.flatMap(row =>
-      m(row).indices.flatMap(col =>
-        createNode((row, col), m(row)(col))
-      )
-    ).mkString("")
-
-  private def createNode(target: (Int, Int), n: SrcNodes): String =
-    if (n.length < 2) ""
-    else {
-      createLink(target, n(0)) + createLink(target, n(1))
+      def createPath(startRow: Int, startCol: Int): String = if (m(startRow)(startCol).isEmpty) "" else
+        s"""  <path
+           |    style='stroke:#000000;fill:none'
+           |    d='M ${startCol * 10 + offsetX},${startRow * 10 + offsetY} $nodeX,$nodeY'
+           |    inkscape:connector-type='polyline'
+           |    inkscape:connector-curvature='0'
+           |    inkscape:connection-start='#${groupId}n${toNodeId(startCol, startRow)}'
+           |    inkscape:connection-end='#${groupId}n${toNodeId(col, row)}'
+           |  />""".stripMargin
+      val node =
+        f"""  <circle
+           |    style='fill:#00${(row % cols + 1) * 32}%2X${(col % rows + 1) * 32}%2X;stroke:none'
+           |    id='${groupId}n${toNodeId(row, col)}'
```

Comment:
Forgot to test with a browser. The links and header line needed fixing, but otherwise its OK. Aha, this line mixed up row and col!

[pattern-sheet.zip](https://github.com/d-bl/GroundForge/files/435664/pattern-sheet.zip)